cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/templates/etl/realtime/ETLWorkerTest.java

```diff
@@ -80,6 +80,18 @@ public static void setupTests() throws IOException {
   }
 
   @Test
+  public void testEmptyProperties() throws Exception {
+    // Set properties to null to test if ETLTemplate can handle it.
+    ETLStage source = new ETLStage("Test", null);
+    ETLStage sink = new ETLStage("Stream", ImmutableMap.of(Properties.Stream.NAME, "testS"));
+    ETLRealtimeConfig etlConfig = new ETLRealtimeConfig(source, sink, Lists.<ETLStage>newArrayList());
+
+    AdapterConfig adapterConfig = new AdapterConfig("null properties", TEMPLATE_ID.getId(), GSON.toJsonTree(etlConfig));
+    Id.Adapter adapterId = Id.Adapter.from(NAMESPACE, "testAdap");
+    createAdapter(adapterId, adapterConfig);
```

Comment:
So there's no verification in this test - just relying on the fact that there were no exceptions? 

Maybe just add something to check that the adapter exists.